### PR TITLE
feat: add IFW rule editor UI for per-component intent filtering

### PR DESCRIPTION
**Pull request by lihenggui**
## Summary
- New `feature:ifwrule` module (api + impl) with per-component IFW rule editor screen
- Scenario-driven UI with user-friendly language — no IFW/XML technical terms exposed to users
- Supports all IFW filter types via intuitive condition cards: action, sender, sender-package, sender-permission, category, host, path, scheme, port, mime-type, data, component pattern
- Bidirectional mapper (`IfwRuleUiMapper`) converts between UI state and `IfwRules` domain model
- Progressive disclosure: simple "block all" toggle for basic use, expandable conditions for advanced users
- Dedicated `IfwRule` per component for independent block/log settings
- Entry point: component long-press context menu → "自定义拦截规则" (hidden for Provider type)
- Advanced options: log toggle, monitor-only mode (block=false)
- Handles unrecognizable legacy rules gracefully with read-only banner

Depends on #1497 (IFW full parameter support)

## Test plan
- [ ] Unit tests for IfwRuleUiMapper (15 tests covering all mapping directions, edge cases, merge strategy)
- [ ] Unit tests for IfwRuleEditorViewModel (5 tests covering load/save/state transitions)
- [ ] Manual test: long press component → "自定义拦截规则" opens editor
- [ ] Manual test: save "block all" → verify component-filter written to IFW file
- [ ] Manual test: save conditional rule → verify and/or/not structure in IFW file
- [ ] Manual test: edit existing rule → verify round-trip preservation
- [ ] Manual test: Provider components should not show the menu item

### CODE REVIEW

File: core/ui/src/main/kotlin/com/merxury/blocker/core/ui/ifwruleeditor/ConditionEditorCard.kt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+@Composable
+internal fun ConditionEditorCard(
+    condition: IfwEditorNode.Condition,
+    depth: Int,
+    onUpdate: (IfwEditorNode) -> Unit,
+    onDelete: (String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    OutlinedCard(
```
Comment (juliewww) [inline core/ui/src/main/kotlin/com/merxury/blocker/core/ui/ifwruleeditor/ConditionEditorCard.kt]: Should use BlockerOutlinedCard()

File: core/ui/src/main/kotlin/com/merxury/blocker/core/ui/ifwruleeditor/GroupEditorCard.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    group: IfwEditorNode.Group,
+    depth: Int,
+    isRoot: Boolean,
+    onUpdate: (IfwEditorNode) -> Unit,
+    onDelete: (String) -> Unit,
+    onAddGroup: (String) -> Unit,
+    onAddCondition: (String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    OutlinedCard(
```
Comment (juliewww) [inline core/ui/src/main/kotlin/com/merxury/blocker/core/ui/ifwruleeditor/GroupEditorCard.kt]: Should use BlockerOutlinedCard()

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    var showUnsavedDialog by remember { mutableStateOf(false) }
+    val canSave = packageName.isNotBlank() && rootGroup.hasTopLevelComponentFilter()
+
+    val handleBack: () -> Unit = {
+        if (isDirty) {
+            showUnsavedDialog = true
+        } else {
+            onBack()
+        }
+    }
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt:85]: The above variables should be managed uniformly in the viewmodel.

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/navigation/GlobalIfwRuleEntryProvider.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import com.merxury.blocker.feature.globalifwrule.impl.GlobalIfwRuleScreen
+import com.merxury.blocker.feature.globalifwrule.impl.GlobalIfwRuleScreenState
+import com.merxury.blocker.feature.globalifwrule.impl.GlobalIfwRuleViewModel
+
+fun EntryProviderScope<NavKey>.globalIfwRuleEntry() {
+    entry<GlobalIfwRuleNavKey> { _ ->
+        val viewModel: GlobalIfwRuleViewModel = hiltViewModel()
+        val editorState = viewModel.editorState.collectAsStateWithLifecycle()
+
+        AnimatedContent(
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/navigation/GlobalIfwRuleEntryProvider.kt]: The logic to judge which screen should not be in the EntryProviderScope, instead of EntryProvider should be concise and own reflect NavKey to the main screen/route.
Create a new main screen/route to deal with the state from the view model and define showing GlobalIfwRuleScreen/AddRuleScreen

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                },
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+            SwitchRow(
+                label = stringResource(R.string.feature_globalifwrule_impl_block),
+                checked = block,
+                onCheckedChange = {
+                    isDirty = true
+                    block = it
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt]: Should wrap into a function in viewmodel

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+                },
+            )
+            SwitchRow(
+                label = stringResource(R.string.feature_globalifwrule_impl_log),
+                checked = log,
+                onCheckedChange = {
+                    isDirty = true
+                    log = it
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt]: Should wrap into a function in viewmodel

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+            Text(
+                text = stringResource(R.string.feature_globalifwrule_impl_conditions),
+                style = MaterialTheme.typography.titleSmall,
+                modifier = Modifier.padding(top = 16.dp, bottom = 8.dp),
+            )
+            IfwRuleTreeEditor(
+                rootGroup = rootGroup,
+                onChange = {
+                    isDirty = true
+                    rootGroup = it
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt]: Should wrap into a function in viewmodel

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+            dismissButton = {
+                BlockerTextButton(onClick = { showUnsavedDialog = false }) {
+                    Text(stringResource(R.string.feature_globalifwrule_impl_cancel))
+                }
+            },
+        )
+    }
+}
+
+data class AddRuleData(
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt]: The dataset class should not be in the UI file, but should be in the view model.

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+                        onSelect(type)
+                        expanded = false
+                    },
+                )
+            }
+        }
+    }
+}
+
+internal val IfwComponentType.labelRes: Int
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/AddRuleScreen.kt]: Should not be in the UI file

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleScreen.kt [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.merxury.blocker.core.designsystem.component.BlockerDropdownMenu
+import com.merxury.blocker.core.designsystem.component.BlockerTextButton
+import com.merxury.blocker.core.designsystem.component.BlockerTopAppBar
+import com.merxury.blocker.core.designsystem.icon.BlockerIcons
+import com.merxury.blocker.core.ui.applist.AppIcon
+import com.merxury.blocker.core.ui.screen.LoadingScreen
+
+@Composable
+fun GlobalIfwRuleScreen(
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleScreen.kt]: Conflicting overloads:
fun GlobalIfwRuleScreen(uiState: GlobalIfwRuleUiState, onAddRuleClick: () -> Unit, onEditRuleClick: (String, Int) -> Unit, onDeleteRule: (String, Int) -> Unit, modifier: Modifier = ...): Unit

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleScreen.kt [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+        uiState = uiState,
+        onAddRuleClick = onAddRuleClick,
+        onEditRuleClick = onEditRuleClick,
+        onDeleteRule = viewModel::deleteRule,
+        modifier = modifier,
+    )
+}
+
+@Composable
+internal fun GlobalIfwRuleScreen(
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleScreen.kt]: Conflicting overloads:
fun GlobalIfwRuleScreen(uiState: GlobalIfwRuleUiState, onAddRuleClick: () -> Unit, onEditRuleClick: (String, Int) -> Unit, onDeleteRule: (String, Int) -> Unit, modifier: Modifier = ...): Unit

And it should not be internal fun

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+    fun saveRule(data: AddRuleData) {
+        if (data.editingRuleIndex != null) {
+            updateRule(data)
+        } else {
+            saveNewRule(data)
+        }
+    }
+
+    private fun saveNewRule(data: AddRuleData) {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt:118]: viewModelScope.launch {} can be replaced by suspend private fun saveNewRule()

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+            componentType = ruleItem.componentType,
+            block = ruleItem.block,
+            log = ruleItem.log,
+            rootGroup = ruleItem.rootGroup,
+            editingRuleIndex = ruleIndex,
+        )
+    }
+
+    private fun updateRule(data: AddRuleData) {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt]: viewModelScope.launch {} can be replaced by suspend private fun updateRule()

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt [hunk 14]
```diff
@@ -0,0 +1,337 @@
...
+                dismissEditor()
+                loadAllRules()
+            } catch (e: Exception) {
+                Timber.e(e, "Failed to update rule for ${data.packageName}")
+            }
+        }
+    }
+
+    fun deleteRule(packageName: String, ruleIndex: Int) {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt:175]: viewModelScope.launch {} can be replaced by suspend private fun deleteRule()

File: feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt [hunk 15]
```diff
@@ -0,0 +1,337 @@
...
+                    loadAllRules()
+                }
+            } catch (e: Exception) {
+                Timber.e(e, "Failed to delete rule for $packageName at index $ruleIndex")
+            }
+        }
+    }
+
+    private fun loadAllRules() {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/globalifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/globalifwrule/impl/GlobalIfwRuleViewModel.kt]: viewModelScope.launch {} can be replaced by suspend private fun loadAllRules()

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/model/RuleEditorUiState.kt [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.merxury.blocker.feature.ifwrule.impl.model
+
+import com.merxury.core.ifw.editor.IfwEditorNode
+import com.merxury.core.ifw.model.IfwComponentType
+
+sealed interface RuleEditorScreenUiState {
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/model/RuleEditorUiState.kt]: Don't need a standalone file, can be in the viewmodel

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+import com.merxury.blocker.core.ui.screen.LoadingScreen
+import com.merxury.blocker.feature.ifwrule.impl.model.BlockMode
+import com.merxury.blocker.feature.ifwrule.impl.model.RuleEditorScreenUiState
+import com.merxury.blocker.feature.ifwrule.impl.model.RuleEditorUiState
+import com.merxury.core.ifw.editor.IfwEditorNode
+
+@Suppress("ktlint:compose:modifier-missing-check")
+@Composable
+fun IfwRuleEditorScreen(
+    snackbarHostState: SnackbarHostState,
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Parameter "snackbarHostState" is never used

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 18]
```diff
@@ -0,0 +1,360 @@
...
+    onBackClick: () -> Unit,
+    viewModel: IfwRuleEditorViewModel,
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val saveComplete by viewModel.saveComplete.collectAsStateWithLifecycle()
+    var isDirty by remember { mutableStateOf(false) }
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Should be managed in the viewmodel

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 19]
```diff
@@ -0,0 +1,360 @@
...
+                }
+            },
+        )
+        when (val state = uiState) {
+            is RuleEditorScreenUiState.Loading -> LoadingScreen()
+            is RuleEditorScreenUiState.Error -> ErrorContent(state.message)
+            is RuleEditorScreenUiState.Success -> EditorContent(
+                editor = state.editor,
+                onUpdateBlockMode = { mode ->
+                    isDirty = true
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: should be wrap into a fun in the viewmodel

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 20]
```diff
@@ -0,0 +1,360 @@
...
+            BlockerTextButton(onClick = onCancel) {
+                Text(stringResource(R.string.feature_ifwrule_impl_cancel))
+            }
+        },
+        modifier = modifier,
+    )
+}
+
+@Composable
+private fun ErrorContent(
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Should use ErrorScreen()
package com.merxury.blocker.core.ui.screen

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 21]
```diff
@@ -0,0 +1,360 @@
...
+    onDelete: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+            .padding(horizontal = 16.dp),
+    ) {
+        // Component name subtitle
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Remove code comments and wrap into a private, standalone function with a meaningful name

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 22]
```diff
@@ -0,0 +1,360 @@
...
+        )
+
+        // Advanced rule banner
+        if (editor.isAdvancedRule) {
+            AdvancedRuleBanner(onDelete = onDelete)
+            Spacer(Modifier.windowInsetsBottomHeight(WindowInsets.safeDrawing))
+            return
+        }
+
+        // Block mode
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Remove code comments and wrap into a private, standalone function with a meaningful name

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 23]
```diff
@@ -0,0 +1,360 @@
...
+            selected = editor.blockMode == BlockMode.ALL,
+            onClick = { onUpdateBlockMode(BlockMode.ALL) },
+        )
+        BlockModeRadioRow(
+            label = stringResource(R.string.feature_ifwrule_impl_block_conditional),
+            selected = editor.blockMode == BlockMode.CONDITIONAL,
+            onClick = { onUpdateBlockMode(BlockMode.CONDITIONAL) },
+        )
+
+        // Conditional section
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Remove code comments and wrap into a private, standalone function with a meaningful name

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 24]
```diff
@@ -0,0 +1,360 @@
...
+                style = MaterialTheme.typography.titleSmall,
+                modifier = Modifier.padding(top = 16.dp, bottom = 8.dp),
+            )
+            IfwRuleTreeEditor(
+                rootGroup = editor.rootGroup,
+                onChange = onUpdateRootGroup,
+            )
+        }
+
+        // Advanced options
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt]: Remove code comments and wrap into a private, standalone function with a meaningful name

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt [hunk 25]
```diff
@@ -0,0 +1,147 @@
...
+
+    private val _saveComplete = MutableStateFlow(false)
+    val saveComplete: StateFlow<Boolean> = _saveComplete.asStateFlow()
+
+    init {
+        loadRules()
+    }
+
+    private fun loadRules() {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt:68]: viewModelScope.launch {} can be replaced by suspend private fun loadRules()

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+        updateEditor { it.copy(blockEnabled = enabled) }
+    }
+
+    fun updateRootGroup(rootGroup: IfwEditorNode.Group) {
+        updateEditor { it.copy(rootGroup = rootGroup) }
+    }
+
+    fun save() {
+        val state = (_uiState.value as? RuleEditorScreenUiState.Success)?.editor ?: return
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt:90]: viewModelScope.launch {} can be replaced by suspend private fun save()

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt [hunk 27]
```diff
@@ -0,0 +1,147 @@
...
+                intentFirewall.saveRules(packageName, merged)
+                _saveComplete.value = true
+            } catch (e: Exception) {
+                Timber.e(e, "Failed to save IFW rules for $packageName")
+            }
+        }
+    }
+
+    fun deleteRule() {
+        viewModelScope.launch {
```
Comment (juliewww) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorViewModel.kt:96]: viewModelScope.launch {} can be replaced by suspend private fun deleteRule()

File: feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt [hunk 28]
```diff
@@ -0,0 +1,368 @@
...
+import com.merxury.blocker.core.designsystem.component.BlockerTopAppBar
+import com.merxury.blocker.core.designsystem.component.BlockerWarningAlertDialog
+import com.merxury.blocker.core.designsystem.icon.BlockerIcons
+import com.merxury.blocker.core.ui.data.UiMessage
+import com.merxury.blocker.core.ui.ifwruleeditor.IfwRuleTreeEditor
+import com.merxury.blocker.core.ui.screen.ErrorScreen
+import com.merxury.blocker.core.ui.screen.LoadingScreen
+import com.merxury.core.ifw.editor.IfwEditorNode
+
+@Suppress("ktlint:compose:modifier-missing-check")
```
Comment (lihenggui) [inline feature/ifwrule/impl/src/main/kotlin/com/merxury/blocker/feature/ifwrule/impl/IfwRuleEditorScreen.kt:61]: Should be removed
